### ISSUE TITLE
Crash in adobe campaign

### ISSUE THREAD

**Issue by VitaliyShuba** [Adobe-Marketing-Cloud/acp-sdks#681]
Hello, we are receiving a lot of crash reports last few days in Adobe campaign
**com.adobe.marketing.mobile:campaign:1.0.9**

```
Fatal Exception: java.util.ConcurrentModificationException:
       at java.util.ArrayList$Itr.next(ArrayList.java:860)
       at com.adobe.marketing.mobile.CampaignExtension.loadConsequences(CampaignExtension.java:679)
       at com.adobe.marketing.mobile.CampaignExtension.onRulesDownloaded(CampaignExtension.java:1233)
       at com.adobe.marketing.mobile.CampaignExtension.downloadRules(CampaignExtension.java:1201)
       at com.adobe.marketing.mobile.CampaignExtension.triggerRulesDownload(CampaignExtension.java:526)
       at com.adobe.marketing.mobile.CampaignExtension.processQueuedEvents(CampaignExtension.java:384)
       at com.adobe.marketing.mobile.CampaignExtension.queueAndProcessEvent(CampaignExtension.java:338)
       at com.adobe.marketing.mobile.CampaignExtension$4.run(CampaignExtension.java:604)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

**Comment by VitaliyShuba** [comment]
Hello, any chance to review this issue?

**Comment by rymorale** [comment]
Hi @VitaliyShuba, thanks for opening the issue. We are working on a fix and will get it released soon.

**Comment by rymorale** [comment]
Hi @VitaliyShuba, the fix for the issue is available in the campaign sdk 1.0.10 release: https://central.sonatype.com/artifact/com.adobe.marketing.mobile/campaign/1.0.10
